src/widgets/fields/baseField.lua

```diff
@@ -0,0 +1,163 @@
+local Runtime = require(script.Parent.Parent.Parent.Runtime)
+local Style = require(script.Parent.Parent.Parent.Style)
+local create = require(script.Parent.Parent.Parent.create)
+local automaticSize = require(script.Parent.Parent.Parent.automaticSize)
+local Field = require(script.Parent.Parent.Parent.Field)
+
+local DEPTH_PADDING = 20
+
+local fieldValueDisplay = Runtime.widget(function(value, options, globalFieldOptions)
+	local fieldValueType = options.fieldValueType or Field.FieldValueType.TEXTBOX
```

Comment:
I don't think that having one master "field" widget and then passing the type in options makes sense. I think that each input should be an entirely separate widget.

Reply:
We already have a checkbox widget, it would be nice if you could just render a field, and then in its children callback, render a checkbox, and have that work as expected

Reply:
Okay that makes sense I'll go ahead and make changes to reflect that. And yes I'm planning on utilizing the existing checkbox widget.